docs/develop/go/test-suites.mdx

```diff
@@ -97,8 +99,281 @@ func TestUnitTestSuite(t *testing.T) {
+
+        err := s.env.GetWorkflowError()
+        s.Error(err)
+        var applicationErr *temporal.ApplicationError
+        s.True(errors.As(err, &applicationErr))
+        s.Equal("SimpleActivityFailure", applicationErr.Error())
+}
+
+func TestUnitTestSuite(t *testing.T) {
+        suite.Run(t, new(UnitTestSuite))
+}
+```
+
+#### Setup
+
+To run unit tests, we first define a test suite struct that absorbs both the
+basic suite functionality from [testify](https://pkg.go.dev/github.com/stretchr/testify/suite)
+via `suite.Suite` and the suite functionality from the Temporal test framework via
+`testsuite.WorkflowTestSuite`. Because every test in this test suite will test our Workflow, we
+add a property to our struct to hold an instance of the test environment. This allows us to initialize
+the test environment in a setup method. For testing Workflows, we use a `testsuite.TestWorkflowEnvironment`.
+
+Next, we implement a `SetupTest` method to set up a new test environment before each test. Doing so
+ensures that each test runs in its own isolated sandbox. We also implement an `AfterTest` function
+where we assert that all the mocks we set up were indeed called by invoking `s.env.AssertExpectations(s.T())`.
+
+Timeout for the entire test can be set using `SetTestTimeout` in the Workflow or Activity environment.
+
+Finally, we create a regular test function recognized by the `go test` command, and pass the struct to `suite.Run`.
+
+#### A Simple Test
+
+The simplest test case we can write is to have the test environment execute the Workflow and then
+evaluate the results.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_Success() {
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_success")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+Calling `s.env.ExecuteWorkflow(...)` executes the Workflow logic and any invoked Activities inside the
+test process. The first parameter of `s.env.ExecuteWorkflow(...)` contains the Workflow functions,
+and any subsequent parameters contain values for custom input parameters declared by the Workflow
+function.
+
+> Note that unless the Activity invocations are mocked or Activity implementation
+> replaced (see [Activity mocking and overriding](#activity-mocking-and-overriding)), the test environment
+> will execute the actual Activity code including any calls to outside services.
+
+After executing the Workflow in the above example, we assert that the Workflow ran through completion
+via the call to `s.env.IsWorkflowComplete()`. We also assert that no errors were returned by asserting
+on the return value of `s.env.GetWorkflowError()`. If our Workflow returned a value, we could have
+retrieved that value via a call to `s.env.GetWorkflowResult(&value)` and had additional asserts on that
+value.
+
+#### Activity mocking and overriding
+
+When running unit tests on Workflows, we want to test the Workflow logic in isolation. Additionally,
```

Comment:
"We want to" is rarely the best framing, imo... if it's not accurate to say "we need to" instead, I'd rewrite this a bit to clarify the benefits of doing it.